Add support for subshell functions

jbrubake opened the pull request:
Functions in the form `func () ( ... )` are documented just like the more common `{}` version.

jbrubake committed:
Add support for subshell functions

Functions in the form `func () ( ... )` are documented just like
the more common `{}` version.

cryobry committed:
Fix underscores in github markdown links

Use the actual header name (do not remove underscores)

Trelor commented on shdoc:
```diff
@@ -835,7 +835,7 @@ match($0, /^([[:blank:]]*#[[:blank:]]+)@(stdin|stdout|stderr)[[:blank:]]+(.*[^[:
 # - `function_name () {`
 # - `function_name {`
-/^[[:blank:]]*(function[[:blank:]]+)?([a-zA-Z0-9_\-:-\\.]+)[[:blank:]]*(\([[:blank:]]*\))?[[:blank:]]*\{/ \
+/^[[:blank:]]*(function[[:blank:]]+)?([a-zA-Z0-9_\-:-\\.]+)[[:blank:]]*(\([[:blank:]]*\))?[[:blank:]]*[({]/ \
```
We don't need to escape it?

`/^[[:blank:]]*(function[[:blank:]]+)?([a-zA-Z0-9_\-:-\\.]+)[[:blank:]]*(\([[:blank:]]*\))?[[:blank:]]*[\{\(]/ \`

jbrubake replied:
Quote the `{` and `(`? No, it's a bracket expression so they are treated literally. You **can** quote them if using `gawk` but I don't think that works as expected if using another `awk`

rafmartom committed:
Fixed typo in README @option instead of @arg

GB609 committed:
Merge pull request #1 from rafmartom/rafmartom-patch-1

Fixed typo in README @option instead of @arg

GB609 committed:
Merge pull request #2 from cryobry/patch-1

Fix underscores in github markdown links

GB609 committed:
Create .editorconfig

GB609 committed:
Rework and enhance `@section`

- Choose header level used for functions depending on section nesting:
  - h2 for top-level
  - h3 for functions contained in sections
  - sections themselves are h2
  - sub-sections coming from annotations will be one level deeper than
the corresponding function
- Introduce `@endsection` to leave active section nesting again.
- `@section` blocks now support `@see`, `@set` and `@example`
- `@section` blocks also appear in TOC, contained functions will be
nested in TOC as well
- Sections are not bound to process_function/render_docblock anymore.
They can also be used when there is no more function following after
them. All that's needed is one none-comment line afterwards.
- added some tests to verify the new behavior

Bugfix:
Reset variable "description" whenever its value was assigned (and thus
consumed) to a `@section` or `@file` block to prevent it from being
inherited/used by the next function as well.

GB609 committed:
Merge pull request #3 from GB609/feature/endsection

Rework and enhance `@section`

- Choose header level used for functions depending on section nesting:
  - h2 for top-level
  - h3 for functions contained in sections
  - sections themselves are h2
  - sub-sections coming from annotations will be one level deeper than
the corresponding function
- Introduce `@endsection` to leave active section nesting again.
- `@section` blocks now support `@see`, `@set` and `@example`
- `@section` blocks also appear in TOC, contained functions will be
nested in TOC as well
- Sections are not bound to process_function/render_docblock anymore.
They can also be used when there is no more function following after
them. All that's needed is one none-comment line afterwards.
- added some tests to verify the new behavior

Bugfix:
Reset variable "description" whenever its value was assigned (and thus
consumed) to a `@section` or `@file` block to prevent it from being
inherited/used by the next function as well.

GB609 committed:
allow to mark entire sections as `@internal`

GB609 committed:
Merge pull request #4 from GB609/feature/internal-sections

allow to mark entire sections as `@internal`

GB609 committed:
fix potentially fatal delete statement

GB609 committed:
allow empty `@file` or `@name` to only get brief and description

GB609 committed:
fix `@file`

GB609 committed:
allow `@see`, `@set` and `@example` for `@file` blocks

GB609 committed:
Update README.md

GB609 committed:
reset docblock_filter after creating file_description

GB609 committed:
fix tests and documentation related to `@file` descriptions

GB609 committed:
new capability: parameter replacements

GB609 committed:
decouple addtional annotations in `@file` blocks from `@description`

GB609 committed:
bugfix: make sure docblock length is only retrieved when its an array

GB609 committed:
Allow raw MD in `@description` blocks (#5)

* introduce literal blocks (no space trimming)
* Create md-literal-lines.test.sh
* Update README.md

GB609 committed:
Update README.md

GB609 committed:
Update README.md
